app/code/community/Oyst/OneClick/controllers/NotificationsController.php

```diff
@@ -48,30 +48,56 @@ public function indexAction()
             return $this->badRequest('No event or data');
         }
 
-        switch ($event) {
-            case 'order.v2.new':
-                $modelName = 'oyst_oneclick/order';
-                break;
-            case 'order.shipments.get':
-                $modelName = 'oyst_oneclick/catalog';
-                break;
-            default:
-                return $this->badRequest('Event name ' . $event . ' is not allow.');
-                break;
-        }
+        if ($event == 'order.v2.new' || $event == 'order.shipments.get') {
+            switch ($event) {
+                case 'order.v2.new':
+                    $modelName = 'oyst_oneclick/order';
+                    break;
+                case 'order.shipments.get':
+                    $modelName = 'oyst_oneclick/catalog';
+                    break;
+                default:
+                    return $this->badRequest('Event name ' . $event . ' is not allow.');
+                    break;
+            }
+
+            try {
+                /** @var Oyst_OneClick_Model_Catalog|Oyst_OneClick_Model_Order $model */
+                $model = Mage::getModel($modelName);
+            } catch (\Exception $e) {
+                return $this->badRequest('Model name ' . $modelName . ' is missing. ' . $e->getMessage());
+            }
 
-        try {
             /** @var Oyst_OneClick_Model_Catalog|Oyst_OneClick_Model_Order $model */
-            $model = Mage::getModel($modelName);
-        } catch (\Exception $e) {
-            return $this->badRequest('Model name ' . $modelName . ' is missing. ' . $e->getMessage());
-        }
+            $response = $model->processNotification($event, $data);
 
-        /** @var Oyst_OneClick_Model_Catalog|Oyst_OneClick_Model_Order $model */
-        $response = $model->processNotification($event, $data);
+            $this->getResponse()->setHeader('Content-type', 'application/json');
+            $this->getResponse()->setBody($response);
+        } elseif ($event == 'notification.newOrder' || $event == 'payment') {
+            if ($event == 'notification.newOrder') {
+                $helperName = 'oyst_oneclick/order_data';
+            } elseif ($event == 'payment') {
+                $helperName = 'oyst_oneclick/payment_data';
+            }
+
+            $helper = Mage::helper($helperName);
```

Comment:
Helper as been refactored into model, so please move ```Helper/Payment/Data.php``` into ```Model/Payment``` and change needed code.